src/main/java/controller/LadderGameController.java

```diff
@@ -0,0 +1,86 @@
+package controller;
+
+import domain.*;
+import java.util.Map;
+import view.InputView;
+import view.OutputView;
+import java.util.List;
+
+public class LadderGameController {
+
+    private static final String END_COMMAND = "all";
+
+    private final InputView inputView;
+    private final OutputView outputView;
+    private BooleanGenerator booleanGenerator;
+
+    private boolean flag = true;
+
+    public LadderGameController(InputView inputView, OutputView outputView, BooleanGenerator booleanGenerator) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+        this.booleanGenerator = booleanGenerator;
+    }
+
+    public void run() {
+        Players players = generatePlayer();
+        Prizes prizes = generatePrize(players.getPlayersSize());
+        Ladder ladder = generateLadder(players);
+        LadderGame ladderGame = new LadderGame(players, ladder, prizes);
+
+        LadderGameResult result = ladderGame.playGame();
+
+        printLadder(players, ladder, prizes);
+        printLadderResultWithRetry(players, result);
+    }
+
+    private Players generatePlayer() {
+        List<String> playerNames = inputView.readPlayerName();
+        Players players = Players.generatePlayer(playerNames);
+        return players;
+    }
+
+    private Prizes generatePrize(int playerCount) {
+        List<String> prizeNames = inputView.readPrizeName();
+        Prizes prizes = Prizes.generatePrizes(playerCount, prizeNames);
+        return prizes;
+    }
+
+    private Ladder generateLadder(Players players) {
+        int ladderHeight = inputView.readLadderHeight();
+        Ladder ladder = Ladder.generateLadder(ladderHeight, players, booleanGenerator);
+        return ladder;
+    }
+
+    private void printLadderResultWithRetry(Players players, LadderGameResult result) {
+        while (flag) {
+            String playerName = inputView.readWantedResultPlayer();
+            printLadderResult(players, result, playerName);
+            isExit(playerName);
```

Comment:
해당 파트는 do-while 문으로 해결할 수 있을 것 같네요 :) 
```suggestion
        String playerName;
        do {
            playerName = inputView.readWantedResultPlayer();
            printLadderResult(players, result, playerName);
        } while (!playerName.equals(END_COMMAND));
```

Reply:
제시하신 코드를 이용해보겠습니다